zummon/src/commonMain/kotlin/companysurvey/Validation.kt

```diff
@@ -768,6 +768,17 @@ val translations: Map<Language, Map<String, Map<String, String>>> = mapOf(
     )
 )
 
+/**
+ * Utility function to display a message in the currently active language.
+ * Intended to replace translation keys, because it's harder to miss a translation,
+ * and it leads to better locality between both translations and between the messages and the code.
+ */
+fun message(nl: String, en: String): String =
```

Comment:
Better option

Reply:
AI can help with the refactoring 

fun message(nl: String, en: String): String = when (getLanguage()) {
    Language.en -> en
    Language.nl -> nl
}

val translations = mapOf(
    "gridConnection" to mapOf(
        "totalPowerChargePointsValid" to message(
            nl = "Totale laadvermogen is geldig",
            en = "Total power of charge points is valid"
        ),
        "totalPowerChargePointsInvalid" to message(
            nl = "Totale laadvermogen %d overschrijdt de toegestane capaciteit %d",
            en = "Total power of charge points %d exceeds allowed capacity %d"
        ),
        "quarterHourlyFeedInLowProductionBatteryPower" to message(
            nl = "Elke kwartierwaarde in feed-in is minder dan of gelijk aan de productie plus de batterijcapaciteit.",
            en = "Every time step in quarter-hourly feed-in is less than or equal to production + battery power"
        ),
        "quarterHourlyFeedInHighProductionBatteryPower" to message(
            nl = "De waarde in dit kwartier van de feed-in (%d) is hoger dan de productie en batterijcapaciteit samen (%d).",
            en = "Time step in quarter-hourly feed-in (%d) is higher than the production and battery power (%d)"
        ),
        "incompatibleQuarterHourly" to message(
            nl = "De kwartierwaarden van de feed-in en productie hebben niet dezelfde lengte.",
            en = "Quarter hourly feed-in is not the same length for Quarter hourly production"
        ),
        "incompatibleStartTimeQuarterHourly" to message(
            nl = "De starttijd van de kwartierwaarden voor feed-in komt niet overeen met die van de productie.",
            en = "Quarter hourly feed-in start time is not the same for Quarter hourly production"
        ),
        "pvInstalledLow" to message(
            nl = "PV geïnstalleerd vermogen blijft lager dan 5000 kW",
            en = "PV installed power stays lower than 5000kW"
        ),
        "pvInstalledHigh" to message(
            nl = "PV geïnstalleerd vermogen mag niet groter zijn dan 5000 kW",
            en = "PV installed power should not be larger than 5000kW"
        )
    ),
    "electricity" to mapOf(
        "kleinverbruikOrGrootverbruikNoDefined" to message(
            nl = "Klein- of grootverbruikstype is niet gedefinieerd",
            en = "Small or large consumption type is not defined"
        ),
        "contractedCapacityNotProvided" to message(
            nl = "Gecontracteerd vermogen ontbreekt",
            en = "Contracted capacity is not provided"
        ),
        "physicalCapacityNotProvide" to message(
            nl = "Fysieke aansluitcapaciteit ontbreekt",
            en = "Physical connection capacity is not provided"
        ),
        "contractedDeliveryCapacityValid" to message(
            nl = "Gecontracteerde aansluitcapaciteit %d is geldig",
            en = "Contracted delivery capacity is valid %d"
        ),
        "contractedDeliveryCapacityExceeds" to message(
            nl = "Gecontracteerde aansluitcapaciteit %d kW overschrijdt fysieke capaciteit %d kW",
            en = "Contracted delivery capacity %d kW exceeds physical capacity %d kW"
        )
    ),
)
request more blocks..

Reply:
Ok I will try this